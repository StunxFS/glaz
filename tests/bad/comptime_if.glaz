fn main() {
    $if (_UNKNOWN_COMPILER_FLAG_) {}
    $if (FLAG & 4) {} // invalid expression
    $if (&FLAG) {}
    $if (FLAG++) {}
}
