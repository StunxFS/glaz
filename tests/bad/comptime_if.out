tests/bad/comptime_if.glaz:2:11: error: unknown compiler flag `_UNKNOWN_COMPILER_FLAG_`
     | 
   2 |     $if (_UNKNOWN_COMPILER_FLAG_) {}
     |          ^^^^^^^^^^^^^^^^^^^^^^^
tests/bad/comptime_if.glaz:3:11: error: non-boolean expression used as comptime `if` condition
     | 
   3 |     $if (FLAG & 4) {} // invalid expression
     |          ^^^^^^^^
tests/bad/comptime_if.glaz:4:11: error: expected `!`, found token `&`
     | 
   4 |     $if (&FLAG) {}
     |          ^^^^^
tests/bad/comptime_if.glaz:5:11: error: unexpected expression for comptime `if` condition
     | 
   5 |     $if (FLAG++) {}
     |          ^^^^^^
glaz error: could not compile `comptime_if`, aborting due to 4 previous errors.
